cla_backend/apps/diagnosis/graph.py

```diff
@@ -137,88 +137,97 @@ def _get_id_default_dict_for(attr, attr_name="attr.name", element="node"):
+        return _text
+
+    def _get_node_data_value_or_default(
+        self, _node, key, as_type=None, translate=True, is_templated=settings.DIAGNOSES_USE_TEMPLATES
+    ):
+        attr_key = self.prop_mapping[key]["id"]
+        try:
+            data_node = self.xpath_ns(_node, 'ns:data[@key="%s"]' % attr_key)[0]
+            value = self._get_text(data_node, is_templated, translate=translate)
+        except IndexError:
+            value = self.prop_mapping[key]["default"]
+        if callable(as_type) and value:
+            return as_type(value)
+        return value
+
+    def _process_context(self, _node, context_key, is_templated):
+        xml_context = self.xpath_ns(_node, 'ns:data[@key="%s"]' % context_key)
+        if not xml_context:
+            return None
+        xml_context = xml_context[0].find("context")
+
+        if xml_context is None:
+            return None
+
+        context = {}
+        for child in xml_context.getchildren():
+            context[child.tag] = self._get_text(child, is_templated)
+        return context
+
+    def process_nodes(self, is_templated=settings.DIAGNOSES_USE_TEMPLATES):
         node_id_map = dict()
         context_key = self.prop_mapping[self.KEY_CONTEXT]["id"]
 
-        def _get_text(a_node, translate=True):
-            if is_templated and a_node.text and "{%" in a_node.text:
-                tpl = get_template_from_string(u"{%% load i18n %%}%s" % a_node.text)
-                if translate:
-                    return lazy(lambda: tpl.render(Context()), text_type)()
-                with translation.override("en"):
-                    return tpl.render(Context())
-            return a_node.text
-
-        def _get_markdown(_text):
-            if is_templated:
-                return lazy(lambda: markdown.markdown(force_text(_text)), text_type)()
-            return _text
-
-        def _process_context(_node):
-            xml_context = self.xpath_ns(_node, 'ns:data[@key="%s"]' % context_key)
-            if not xml_context:
-                return None
-            xml_context = xml_context[0].find("context")
-
-            if xml_context is None:
-                return None
-
-            context = {}
-            for child in xml_context.getchildren():
-                context[child.tag] = _get_text(child)
-            return context
-
-        def _get_node_data_value_or_default(_node, key, as_type=None, translate=True):
-            attr_key = self.prop_mapping[key]["id"]
-            try:
-                data_node = self.xpath_ns(_node, 'ns:data[@key="%s"]' % attr_key)[0]
-                value = _get_text(data_node, translate=translate)
-            except IndexError:
-                value = self.prop_mapping[key]["default"]
-            if callable(as_type) and value:
-                return as_type(value)
-            return value
-
         def str_to_bool(s):
             return s.lower() == "true"
```

Comment:
Random suggestion but this can be made into a static method?

Reply:
Yep, moved it up to class function now